### ISSUE TITLE
foreignAutoCreate dont create inner object

### ISSUE THREAD

**Issue by Knight704** [j256/ormlite-core#46]
I use robospice-ormlite version 1.4.14, which depends on ormlite-android 4.48 and ormlite-core 4.48.
Like the documentation said for foreignAutoCreate: "Set this to be true (default false) to have the foreign field automatically created using an internal DAO if its ID field is not set (null or 0)."

I used it like this:
```java
@DatabaseTable(tableName = "parent")
public class Parent {
    @DatabaseField(columnName = "_id", generatedId = true)
    public long databaseId;

    @DatabaseField(columnName = "foreignId", foreign = true, foreignAutoCreate = true,
            foreignAutoRefresh = true, foreignColumnName = "child_api_id", canBeNull = false)
    public Child child;
...
}

@DatabaseTable(tableName = "child")
public class Child {
    @DatabaseField(columnName = "_id", generatedId = true)
    public long databaseId;

    @DatabaseField(columnName = "api_id", columnDefinition = "INTEGER UNIQUE ON CONFLICT REPLACE")
    public long child_api_id;
...
}
```
Parent object retrieved from json, so its child fields already instantiated, except databaseId, which is zero of course. I thought that, when i'll persist the parent, childs are also will be persisted because the databaseId field is zero. But it is not persisted, probably because the child_api_id not zero? Nonetheless the child_api_id are correctly persisted to Parent table in column "foreignId".

The reason i cannot use a _id field of Child as foreign, that it will change over time because of UNIQUE ON CONFLICT REPLACE. Also i cannot persist the Child before Parent manually, because this job is done by CacheManager of RoboSpice.

Best regards, Dmitriy.

**Comment by Knight704** [comment]
Ok, finally i found the temporal solution: use the own DAO class and manually persist all object i need. To do so, provide which DAO to use to your databaseTable annotation.

Though the foreignAutoCreate parameter are still behave not as i expected, maybe this will be helpfull someone.

**Comment by j256** [comment]
This is old, sorry, but I can't reproduce this.  Here's the test I just added.  I removed the columnDefinition and had to change the `foreignColumnName` to match the `child_api_id` `columnName`.  See: https://github.com/j256/ormlite-core/commit/9f34c7371c580ba0d138176265490df904702fcf

**Comment by j256** [comment]
Ok.  I now understand this problem and it is a bug specifically around the foreignColumnName and different field types.  This will be fixed in 5.0.
